<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Confidence-Intervals" data-toc-modified-id="Confidence-Intervals-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Confidence Intervals</a></span><ul class="toc-item"><li><span><a href="#Challenge-1" data-toc-modified-id="Challenge-1-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Challenge 1</a></span></li><li><span><a href="#Challenge-2" data-toc-modified-id="Challenge-2-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Challenge 2</a></span></li><li><span><a href="#Challenge-3---More-practice" data-toc-modified-id="Challenge-3---More-practice-1.3"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Challenge 3 - More practice</a></span></li><li><span><a href="#Challenge-4---More-practice" data-toc-modified-id="Challenge-4---More-practice-1.4"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>Challenge 4 - More practice</a></span></li><li><span><a href="#Bonus-Challenge" data-toc-modified-id="Bonus-Challenge-1.5"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>Bonus Challenge</a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#1.--Determine-what-size-the-selected-sample-of-men-should-be." data-toc-modified-id="1.--Determine-what-size-the-selected-sample-of-men-should-be.-1.5.0.1"><span class="toc-item-num">1.5.0.1&nbsp;&nbsp;</span>1.- Determine what size the selected sample of men should be.</a></span></li><li><span><a href="#2.--For-the-second-challenge,-we-have-the-following-error:" data-toc-modified-id="2.--For-the-second-challenge,-we-have-the-following-error:-1.5.0.2"><span class="toc-item-num">1.5.0.2&nbsp;&nbsp;</span>2.- For the second challenge, we have the following error:</a></span></li><li><span><a href="#Determine-the-sample-size-required-to-not-exceed-an-error-of-1%-with-a-confidence-of-80%." data-toc-modified-id="Determine-the-sample-size-required-to-not-exceed-an-error-of-1%-with-a-confidence-of-80%.-1.5.0.3"><span class="toc-item-num">1.5.0.3&nbsp;&nbsp;</span>Determine the sample size required to not exceed an error of 1% with a confidence of 80%.</a></span></li></ul></li></ul></li><li><span><a href="#Bonus-Challenge" data-toc-modified-id="Bonus-Challenge-1.6"><span class="toc-item-num">1.6&nbsp;&nbsp;</span>Bonus Challenge</a></span></li></ul></li></ul></div>

# Confidence Intervals

In [3]:
pip install scipy

                                              0.0/44.0 MB ? eta -:--:--
                                              0.3/44.0 MB 7.9 MB/s eta 0:00:06
     -                                        1.4/44.0 MB 17.2 MB/s eta 0:00:03
     --                                       2.9/44.0 MB 22.8 MB/s eta 0:00:02
     ----                                     4.6/44.0 MB 26.6 MB/s eta 0:00:02
     -----                                    6.2/44.0 MB 28.3 MB/s eta 0:00:02
     -------                                  8.1/44.0 MB 32.2 MB/s eta 0:00:02
     ---------                               10.2/44.0 MB 34.2 MB/s eta 0:00:01
     ----------                              12.3/44.0 MB 40.9 MB/s eta 0:00:01
     ------------                            14.5/44.0 MB 43.5 MB/s eta 0:00:01
     --------------                          16.7/44.0 MB 46.9 MB/s eta 0:00:01
     ----------------                        18.8/44.0 MB 46.9 MB/s eta 0:00:01
     ------------------                      21.0


[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: C:\Users\andys\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


In [5]:
# Libraries
from scipy.stats import norm
import numpy as np

## Challenge 1
We want to estimate the average size of the men of a country with a confidence level of 80%. Assuming that the standard deviation of the sizes in the population is 4, get the confidence interval with a sample of men selected randomly, whose heights are:

````
heights = [167, 167, 168, 168, 168, 169, 171, 172, 173, 175, 175, 175, 177, 182, 195]
````

**Hint**: function `stats.norm.interval` from `scipy` can help you get through this exercise. 

In [12]:
# your code here
heights = [167, 167, 168, 168, 168, 169, 171, 172, 173, 175, 175, 175, 177, 182, 195]
mu = np.mean(heights)
sigma = np.std(heights)
norm.interval(0.8, loc=mu, scale=sigma)

(164.35812913248898, 182.57520420084435)

In [11]:
norm.interval(0.8, loc=mean, scale=sigma / np.sqrt(len(heights)))

(171.11485238812165, 175.8184809452117)

## Challenge 2 
In a sample of 105 shops selected randomly from an area, we note that 27 of them have had losses in this month. Get an interval for the proportion of businesses in the area with losses to a confidence level of 80% and a confidence level of 90%.

**Hint**: function `stats.norm.interval` from `scipy` can help you get through this exercise. 

In [ ]:
# your code here

## Challenge 3 - More practice
For the same example in challenge 1, calculate a confidence interval for the variance at 90% level.

**Hint**: function `stats.chi2.interval` from `scipy` can help you get through this exercise. 

In [1]:
# your code here

## Challenge 4 - More practice
The sulfuric acid content of 7 similar containers is 9.8, 10.2, 10.4, 9.8, 10.0, 10.2 and 9.6 liters. Calculate a 95% confidence interval for the average content of all containers assuming an approximately normal distribution.

```
acid = [9.8, 10.2, 10.4, 9.8, 10.0, 10.2, 9.6]
```

**Hint**: function `stats.t.interval` from `scipy` can help you get through this exercise. 

In [ ]:
# your code here

## Bonus Challenge
The error level or sampling error for the first challenge is given by the following expression:
$$Error = z_{\frac{\alpha}{2}}\frac{\sigma}{\sqrt n}$$
Where z represents the value for N(0,1)


Suppose that with the previous data of challenge 1, and with a confidence level of
99% (that is, almost certainly) we want to estimate the average population size, so that the error level committed is not greater than half a centimeter.

#### 1.- Determine what size the selected sample of men should be.

In [ ]:
# your code here

#### 2.- For the second challenge, we have the following error:
$$ Error = z_{\frac{\alpha}{2}}\sqrt{\frac{p\times q}{n}} $$
#### Determine the sample size required to not exceed an error of 1% with a confidence of 80%.

In [ ]:
# your code here

## Bonus Challenge

Let's consider the following problem:

Build a confidence interval of 94% for the real difference between the durations of two brands of spotlights, if a sample of 40 spotlights taken randomly from the first mark gave an average duration of 418 hours, and a sample of 50 bulbs of another brand gave a duration average of 402 hours. The standard deviations of the two
populations are 26 hours and 22 hours, respectively.

Sometimes, we will be interested in the difference of two different groups of random variables. We can also build a confidence interval for that! We have some different cases regarding the variance but for this specific case (the variance are different and known), we have that:

$$\overline{X} - \overline{Y}  \sim N(\mu_{X} - \mu_{Y} , \sqrt{\frac{\sigma_{X}^2}{n_X}+\frac{\sigma_{Y}^2}{n_Y}})$$

Solve the problem with this information.

In [ ]:
# your code here